internal/engine/buildcontrol/build_control.go

```diff
@@ -21,7 +21,7 @@ func NextTargetToBuild(state store.EngineState) *store.ManifestTarget {
 
 	// Local targets aren't parallelizable with any other kind of target.
 	// So if we're already building a local target, bail immediately.
-	if IsBuildingLocalTarget(state) {
+	if IsBuildingUnparallelizableLocalTarget(state) {
```

Comment:
unless I've misunderstood the intent of this PR, we also need to change the `RemoveLocalTargets` bit a few lines down

Reply:
oooh good catch! fixed and added a test